### PR TITLE
Forge Github Org: Use `login` instead of `name`

### LINKED ISSUE #2092: Enabling second repository results in: `pq: duplicate key value violates unique constraint "UQE_orgs_name"`
### Component

server

### Describe the bug

Enabling my first repository worked fine, however the second does not.

When the UI sends a POST request to https://woodpecker-host/api/repos?forge_remote_id=526933511 it returns this error
```
pq: duplicate key value violates unique constraint "UQE_orgs_name"
```

The debug log of the server does not give any more insights:
```
{ip: 10.100.32.116, latency: 474.729763, level: debug, method: POST, path: /api/repos, status: 500, user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36}
```

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"1.0.0"}
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### PR DESCRIPTION
**Pull request by runephilosof-karnovgroup**
For "woodpecker-ci" the `name` is "Woodpecker CI"
 and the `login` is "woodpecker-ci"

Fixes #2092

This was causing the organization lookup to fail, because it looks up using the `login`, when it did not find the organization, it would try to create it. The creation would fail, because it uses the `name`, and an organization with that `name` already exists.
Resulting in
```
pq: duplicate key value violates unique constraint "UQE_orgs_name"
```

### CODE REVIEW

**Comment by runephilosof-karnovgroup** [comment]
The commit is based on the merge-base of main and release/v1.0 to make it easy to also merge it into release/v1.0

**Comment by runephilosof-karnovgroup** [comment]
> Can you check this for Gitlab, Gitea and Bitbucket too?

Looking at the code for Gitlab it seems problematic.
While Gitea and Bitbucket seems fine.

The foolproof way would be to use the `owner` variable instead of what the forge returns.
However, we are basing `IsUser` and `Private` on what we look up in the forge, so it is important that we handle the response from the forge correct.

File: server/forge/github/github.go [hunk 1]
```diff
@@ -354,20 +354,22 @@ func (c *client) Org(ctx context.Context, u *model.User, owner string) (*model.O
 	client := c.newClientToken(ctx, u.Token)
 
 	user, _, err := client.Users.Get(ctx, owner)
+	log.Trace().Msgf("Github user for owner %s = %v", owner, user)
```
Comment (runephilosof-karnovgroup) [inline server/forge/github/github.go:357]: Do you like this trace logging. Or should I remove it?

**Comment by runephilosof-karnovgroup** [comment]
I have fixed it locally for Gitlab as well now. It should be using `FullPath`.
I will create a new PR for fixing Gitlab.

**Comment by runephilosof-karnovgroup** [comment]
On second thought, I will include the Gitlab fix here, because both concerns the same issue #2092.

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
Backported -> #2106

**Pull request merged**